### PR TITLE
bpo-30544: _io._WindowsConsoleIO.write raises the wrong error when WriteConsoleW fails

### PR DESCRIPTION
**Pull request by segevfiner**
Found due to this: https://github.com/pytest-dev/py/issues/103

### CODE REVIEW

File: Modules/_io/winconsoleio.c [hunk 1]
```diff
@@ -1000,6 +1001,9 @@ _io__WindowsConsoleIO_write_impl(winconsoleio *self, Py_buffer *b)
     wlen = MultiByteToWideChar(CP_UTF8, 0, b->buf, len, wbuf, wlen);
     if (wlen) {
         res = WriteConsoleW(self->handle, wbuf, wlen, &n, NULL);
+        if (!res)
```
Comment (eryksun) [inline Modules/_io/winconsoleio.c]: There's no point in calculating `len` if the call failed, so why not instead add this to the following test, i.e. `(res && n < wlen)`?

**Review by zooba — APPROVED** [review]
Looks good to me. Ignore the AppVeyor failure - it is unrelated.

**Comment by zooba** [comment]
Forgot that I can just hit merge (different processes for me on different projects).

Still needs a backport - I'll try to get to it today, but if someone else cherrypicks the change into 3.6 and submits as a PR, mention me and I'll merge.

**Comment by segevfiner** [comment]
@zooba https://github.com/python/cpython/pull/1920 for the AppVeyor failure.

**Comment by zooba** [comment]
@segevfiner Merged that one too. Thanks!

**Pull request merged**